utilities/MasImuTest/src/MasImuTest.cpp

```diff
@@ -0,0 +1,855 @@
+
+    m_previousPositionFeedbackInRad = m_positionFeedbackInRad;
+
+    return true;
+}
+
+bool MasImuTest::MasImuData::addSample()
+{
+    if (isCompleted())
+    {
+        return true;
+    }
+
+    std::string errorPrefix = "[MasImuTest::MasImuData::addSample](" + m_testName +") ";
+
+    bool ok = getFeedback();
+    if (!ok)
+    {
+        yError() << errorPrefix << "Failed to get feedbacks.";
+        return false;
+    }
+
+    if (maxVariation() < m_commonDataPtr->minJointVariationRad)
+    {
+        return true;
+    }
+
+    ok = updateRotationFromEncoders();
+    if (!ok)
+    {
+        yError() << errorPrefix << "updateRotationFromEncoders() failed.";
+        return false;
+    }
+
+    iDynTree::Rotation measuredImu = m_imuWorld * m_rotationFeedback;
+
+    if (m_commonDataPtr->filterYaw)
+    {
+        double measuredRoll, measuredPitch, measuredYaw;
+        measuredImu.getRPY(measuredRoll, measuredPitch, measuredYaw);
+
+        double estimatedRoll, estimatedPitch, estimatedYaw;
+        m_rotationFromEncoders.getRPY(estimatedRoll, estimatedPitch, estimatedYaw);
+
+        measuredImu = iDynTree::Rotation::RPY(measuredRoll, measuredPitch, estimatedYaw);
+    }
+
+    iDynTree::Rotation error = m_rotationFromEncoders.inverse() * measuredImu;
+
+    m_data.push_back(error);
+
+    m_previousPositionFeedbackInRad = m_positionFeedbackInRad;
+
+    yInfo() << errorPrefix << "Sample " << addedSamples() << "/" << m_commonDataPtr->maxSamples;
+
+    if (static_cast<int>(addedSamples()) >= m_commonDataPtr->maxSamples)
+    {
+        setCompleted();
+    }
+
+    return true;
+}
+
+size_t MasImuTest::MasImuData::addedSamples() const
+{
+    return m_data.size();
+}
+
+bool MasImuTest::MasImuData::isCompleted() const
+{
+    return m_completed;
+}
+
+void MasImuTest::MasImuData::setCompleted()
+{
+    m_completed = true;
+}
+
+void MasImuTest::MasImuData::printResults() const
+{
+    std::string errorPrefix = "[MasImuTest::MasImuData::printResults](" + m_testName +") ";
+
+    if (!m_data.size())
+    {
+        yInfo() << errorPrefix << "Results ("<<m_data.size() << " samples) :\n"
+                << "--------------------------------------\n"
+                << "--------------------------------------\n";
+        return;
+    }
+
+    iDynTree::Rotation meanError;
+
+    iDynTree::GeodesicL2MeanOptions options;
+    options.maxIterations = 1000; //If it takes so many steps to converge, it is probably in a loop. Hence, this parameter can be safely hardcoded.
+
+    bool ok = iDynTree::geodesicL2MeanRotation(m_data, meanError, options);
+    if (!ok)
+    {
+        yError() << errorPrefix << "Failed to compute the mean rotation.";
+        return;
+    }
+
+    double minError = - 1;
+    double maxError = -1;
+    size_t minIndex, maxIndex;
+
+    for (size_t i = 0; i < m_data.size(); ++i)
+    {
+        double error = iDynTree::geodesicL2Distance(iDynTree::Rotation::Identity(), m_data[i]);
```

Comment:
It's the same as directly applying `toEigen(m_data[i].log()).norm()` in case you want to avoid an additional function call.

Reply:
I think it is more clear with the ``iDynTree`` code TBH :thinking:

Reply:
In that case, ok. However, the underlying difference is just a composition of `m_data[i]` with the identity matrix, which seemed redundant.

Reply:
Yes, I wrote it assuming not to know anything of the underlying method :grin:

Reply:
It doesn't hurt computationally. So we go with the iDynTree code.